packages/flutter/lib/src/widgets/animated_size.dart

```diff
@@ -100,14 +103,57 @@ class AnimatedSize extends SingleChildRenderObjectWidget {
   /// If not specified, defaults to [duration].
   final Duration? reverseDuration;
 
-  /// The [TickerProvider] for this widget.
-  final TickerProvider vsync;
-
   /// {@macro flutter.material.Material.clipBehavior}
   ///
   /// Defaults to [Clip.hardEdge], and must not be null.
   final Clip clipBehavior;
 
+  @override
+  _AnimatedSizeState createState() => _AnimatedSizeState();
+}
+
+class _AnimatedSizeState
+    extends State<AnimatedSize> with SingleTickerProviderStateMixin {
+  @override
+  Widget build(BuildContext context) {
+    return _AnimatedSize(
+      child: widget.child,
+      alignment: widget.alignment,
+      curve: widget.curve,
+      duration: widget.duration,
+      reverseDuration: widget.reverseDuration,
+      vsync: this,
+      clipBehavior: widget.clipBehavior,
+    );
+  }
+}
+
+class _AnimatedSize extends SingleChildRenderObjectWidget {
+  /// Creates a widget that animates its size to match that of its child.
```

Comment:
this doc can probably be removed since the constructor is private and this doesn't contain much information...

Reply:
done